src/blockdata/transaction.rs

```diff
@@ -83,6 +122,22 @@ impl Default for OutPoint {
     }
 }
 
+impl ::std::str::FromStr for OutPoint {
+    type Err = ParseOutPointError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let find = s.find(':');
+        if find == None || find != s.rfind(':') {
```

Comment:
Creative :).

Can we add a check that the total length does not exceed 75 (64 bytes txid plus one `:` plus 10 bytes decimal number) before calling either `find`? Then the total performance is bounded even for giant pathological inputs.

Reply:
Addressed in extra commit.

Reply:
10 digits covers the whole `u32` range. I don't think we need to be more clever than that.